versioned/persist/adapter/src/main/java/org/projectnessie/versioned/persist/adapter/spi/AbstractDatabaseAdapter.java

```diff
@@ -1280,6 +1313,7 @@ protected Hash copyCommits(
 
     int keyListDistance = targetHeadCommit != null ? targetHeadCommit.getKeyListDistance() : 0;
 
+    List<CommitLogEntry> unwrittenEntriesForKeyList = new LinkedList<>();
```

Comment:
Still - no need for an expensive LinkedList here. ArrayList is sufficient

Reply:
changed.